### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by CyberDem0n**
* document tags
* move dynamic configuration out of `bootstrap.dcs`
* document REST API endpoints

### CODE REVIEW

**Comment by sdudoladov** [comment]
:+1:

File: docs/SETTINGS.rst [hunk 1]
```diff
@@ -4,6 +4,39 @@
...
 
+.. _dynamic_configuration_settings:
+
+Dynamic configuration settings
+------------------------------
+
+Dynamic configuration is stored in the DCS (Distributed Configuration Store) and applied on all cluster nodes. Some parameters, like **loop_wait**, **ttl**, **postgresql.parameters.max_connections**, **postgresql.parameters.max_worker_processes** and so on could be set only in the dynamic configuration. Some other parameters like **postgresql.listen**, **postgresql.data_dir** could be set only locally, i.e. in the Patroni config file or via :ref:`configuration <environment>` variable. In most cases the local configuration will override the dynamic configuration. In order to change the dynamic configuration you can use either ``patronictl edit-config`` tool or Patroni :ref:`REST API <rest_api>`.
+
+-  **loop\_wait**: the number of seconds the loop will sleep. Default value: 10
+-  **ttl**: the TTL to acquire the leader lock. Think of it as the length of time before initiation of the automatic failover process. Default value: 30
```
Comment (avaczi) [inline docs/SETTINGS.rst]: the TTL to acquire the leader lock _in seconds_

File: docs/SETTINGS.rst [hunk 2]
```diff
@@ -4,6 +4,39 @@
...
+-  **retry\_timeout**: timeout for DCS and PostgreSQL operation retries. DCS or network issues shorter than this will not cause Patroni to demote the leader. Default value: 10
```
Comment (avaczi) [inline docs/SETTINGS.rst]: like above

File: docs/SETTINGS.rst [hunk 3]
```diff
@@ -4,6 +4,39 @@
...
+-  **maximum\_lag\_on\_failover**: the maximum bytes a follower may lag to be able to participate in leader election.
+-  **master\_start\_timeout**: the amount of time a master is allowed to recover from failures before failover is triggered. Default is 300 seconds. When set to 0 failover is done immediately after a crash is detected if possible. When using asynchronous replication a failover can cause lost transactions. Best worst case failover time for master failure is: loop\_wait + master\_start\_timeout + loop\_wait, unless master\_start\_timeout is zero, in which case it's just loop\_wait. Set the value according to your durability/availability tradeoff.
```
Comment (avaczi) [inline docs/SETTINGS.rst]: (in seconds). Default is 300.

Then later: 'Best worst' from which is worst should stay.

File: docs/SETTINGS.rst [hunk 4]
```diff
@@ -4,6 +4,39 @@
...
+-  **synchronous\_mode**: turns on synchronous replication mode. In this mode a replica will be chosen as synchronous and only the latest leader and synchronous replica are able to participate in leader election. Synchronous mode makes sure that successfully committed transactions will not be lost at failover, at the cost of losing availability for writes when Patroni cannot ensure transaction durability. See :ref:`replication modes documentation <replication_modes>` for details.
+-  **synchronous\_mode\_strict**: prevents disabling synchronous replication if no synchronous replicas are available, blocking all client writes to the master. See :ref:`replication modes documentation <replication_modes>` for details.
+-  **postgresql**:
+    -  **use\_pg\_rewind**: whether or not to use pg_rewind
+    -  **use\_slots**: whether or not to use replication_slots.
+    -  **recovery\_conf**: additional configuration settings written to recovery.conf when configuring follower.
```
Comment (avaczi) [inline docs/SETTINGS.rst]: Maybe worth mentioning this has changed with PG 12?

File: docs/SETTINGS.rst [hunk 5]
```diff
@@ -27,32 +60,7 @@ Log
...
-            -  **type**: slot type. Could be ``physical`` or ``logical``. If the slot is logical, you have to additionally define ``database`` and ``plugin``.
-            -  **database**: the database name where logical slots should be created.
-            -  **plugin**: the plugin name for the logical slot.
+-  **dcs**: This section will be written into `/<namespace>/<scope>/config` of a given configuration store after initializing of new cluster. The global dynamic configuration for the cluster. Under the ``bootstrap.dcs`` you can put any of the parameters described in the :ref:`Dynamic Configuration settings <dynamic_configuration_settings>` and after Patroni initialized (bootstrapped) the new cluster it will write this section into `/<namespace>/<scope>/config` of a given configuration store. All later changes of ``bootstrap.dcs`` will not take any effect! If you want to change them please use either ``patronictl edit-config`` or Patroni :ref:`REST API <rest_api>`.
```
Comment (avaczi) [inline docs/SETTINGS.rst]: of _the_ given
the new cluster**,** it will 
of the configuration store.

File: docs/SETTINGS.rst [hunk 6]
```diff
@@ -190,7 +202,7 @@ PostgreSQL
...
+-  **connect\_address**: IP address (or hostname) and port, to access the Patroni's :ref:`REST API <rest_api>`. All the members of the cluster must be able to connect to this address, so unless the Patroni setup is intended for a demo inside the localhost, this address must be a non "localhost" or loopback addres (ie: "localhost" or "127.0.0.1"). It can serve as a endpoint for HTTP health checks (read below about the "listen" REST API parameter), and also for user queries (either directly or via the REST API), as well as for the health checks done by the cluster members during leader elections (for example, to determine whether the master is still running, or if there is a node which has a WAL position that is ahead of the one doing the query; etc.) The connect_address is put in the member key in DCS, making it possible to translate the member name into the address to connect to its REST API.
```
Comment (avaczi) [inline docs/SETTINGS.rst]: loopback addres_s_
a_n_ endpoint

File: docs/dynamic_configuration.rst [hunk 7]
```diff
@@ -14,7 +14,7 @@ Patroni configuration is stored in the DCS (Distributed Configuration Store). Th
 
 - Local :ref:`configuration <settings>` (patroni.yml).
 	These options are defined in the configuration file and take precedence over dynamic configuration.
-	patroni.yml could be changed and reload in runtime (without restart of Patroni) by sending SIGHUP to the Patroni process or by performing ``POST /reload`` REST-API request.
+	patroni.yml could be changed and reload in runtime (without restart of Patroni) by sending SIGHUP to the Patroni process, performing ``POST /reload`` REST-API request or executing ``patronictl reload``.
```
Comment (avaczi) [inline docs/dynamic_configuration.rst]: reload_ed_

File: docs/rest_api.rst [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+
+- ``POST /restart``: You can restart Postgres on the specific node by performing the ``POST /restart`` call. In the JSON body of ``POST`` request it is possible to optionally specify some restart conditions:
+
+  - **restart_pending**: boolean, if set to ``true`` Patroni will restart PostgreSQL only when restart is pending in order to apply some changes in the PostgreSQL config.
+  - **role**: perform restart only if the current role of the node matches with the role from the POST request.
+  - **postgres_version**: perform restart only if the current version of postgres is smaller than specified in the POST request.
+  - **timeout**: how long we should wait before PostgreSQL starts accepting connections. Overrides ``master_start_timeout``.
+  - **schedule**: timestamp with time zone, schedule the restart somewhere in the future.
+
+- ``DELETE /restart``: delete the scheduled restart
```
Comment (avaczi) [inline docs/rest_api.rst]: this is used with ``patronictl flush``, right?

Comment (CyberDem0n) [reply docs/rest_api.rst]: Yes, updated and pushed.

**Comment by avaczi** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**